src/components/core/tooltip/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,172 @@
+    '100%': { opacity: 1, transform: 'translateY(0)' },
+});
+
+const slideRightAndFade = keyframes({
+    '0%': { opacity: 0, transform: 'translateX(-50%)' },
+    '100%': { opacity: 1, transform: 'translateX(0)' },
+});
+
+const slideDownAndFade = keyframes({
+    '0%': { opacity: 0, transform: 'translateY(-50%)' },
+    '100%': { opacity: 1, transform: 'translateY(0)' },
+});
+
+const slideLeftAndFade = keyframes({
+    '0%': { opacity: 0, transform: 'translateX(50%)' },
+    '100%': { opacity: 1, transform: 'translateX(0)' },
+});
+
+const StyledContent = styled(TooltipPrimitive.Content, {
+    padding: '0.5rem',
+    fontSize: '$3xs',
+    lineHeight: '$lineHeight18',
+    borderRadius: 4,
+    userSelect: 'none',
+    '@media (prefers-reduced-motion: no-preference)': {
+        animationDuration: '0.18s',
+        animationTimingFunction: 'ease-out',
+        willChange: 'transform, opacity',
+        '&[data-state="delayed-open"]': {
+            '&[data-side="top"]': { animationName: slideUpAndFade },
+            '&[data-side="right"]': { animationName: slideRightAndFade },
+            '&[data-side="bottom"]': { animationName: slideDownAndFade },
+            '&[data-side="left"]': { animationName: slideLeftAndFade },
+        },
+    },
+    '@mobile': {
+        fontSize: '$4xs',
+        lineHeight: '$lineHeight14',
+    },
+    variants: {
+        type: {
+            default: {
+                backgroundColor: '$greyLight400',
+                color: '$greyLight700',
+            },
+            error: {
+                backgroundColor: '$redLight',
+                color: '$greyLight100',
+            },
+        },
+        dark: {
+            true: {
+                backgroundColor: '$greyDark400',
+                color: '$greyDark100',
+            },
+        },
+    },
+    compoundVariants: [
+        {
+            dark: true,
+            type: 'error',
+            css: {
+                backgroundColor: '$redDark',
+                color: '$greyLight100',
+            },
+        },
+    ],
+});
+
+const StyledArrow = styled(TooltipPrimitive.Arrow, {
+    width: 12,
+    height: 6,
+    fill: '$greyLight400',
+    variants: {
+        type: {
+            default: {
+                fill: '$greyLight400',
+            },
+            error: {
+                fill: '$redLight',
+            },
+        },
+        dark: {
+            true: {
+                fill: '$greyDark400',
+            },
+        },
+    },
+    compoundVariants: [
+        {
+            dark: true,
+            type: 'error',
+            css: {
+                fill: '$redDark',
+            },
+        },
+    ],
+});
+
+const Tooltip = ({ children, type = 'error', tooltip_content, icon, is_fixed_width, side = 'top' }: TooltipProps) => {
+    const { isDark } = useTheme();
+    const [open_tooltip, setOpenTooltip] = useState(false);
+    return (
+        <TooltipPrimitive.Provider>
```

Comment:
@vinu-deriv Shouldn’t we wrap the whole app with the provider instead of the component itself? 🤔

Reply:
from deriv-ui only components are exported and used in other projects. Tooltip needs provider for its functionality, in that case the tooltip.provider must be within tooltip component for tooltip to work.

Reply:
@vinu-deriv as stated [here](https://www.radix-ui.com/docs/primitives/components/tooltip#provider) we have to wrap the whole app inside `TooltipProvider`. with this approach every time someone uses the tooltip component there will be a new `context` for it as well. 
can we try to use the global approach and provide needed functions to change the `dealy`, etc ?